__init__.py

```diff
@@ -7,77 +7,42 @@
 sys.path.append('/apps/digiclk/')
 import monotime as utime
 import draw
+import nicesegments
+import config
+import battery
+from globals import *
 
-DIGITS = [
-    (True, True, True, True, True, True, False),
-    (False, True, True, False, False, False, False),
-    (True, True, False, True, True, False, True),
-    (True, True, True, True, False, False, True),
-    (False, True, True, False, False, True, True),
-    (True, False, True, True, False, True, True),
-    (True, False, True, True, True, True, True),
-    (True, True, True, False, False, False, False),
-    (True, True, True, True, True, True, True),
-    (True, True, True, True, False, True, True)
-]
-
-def renderNum(d, num, x):
-    draw.Grid7Seg(d, x, 0, 7, DIGITS[num // 10], (255, 255, 255))
-    draw.Grid7Seg(d, x + 5, 0, 7, DIGITS[num % 10], (255, 255, 255))
-
-def renderColon(d):
-    draw.GridVSeg(d, 11, 2, 7, 2, (255, 255, 255))
-    draw.GridVSeg(d, 11, 4, 7, 2, (255, 255, 255))
+THEMES = [draw, nicesegments]
 
 def renderText(d, text, blankidx = None):
     bs = bytearray(text)
 
     if blankidx != None:
         bs[blankidx:blankidx+1] = b'_'
 
-    d.print(MODES[MODE] + ' ' + bs.decode(), fg = (255, 255, 255), bg = None, posx = 0, posy = 7 * 8)
-
-def renderBar(d, num):
-    d.rect(20, 78, 20 + num * 2, 80, col = (255, 255, 255))
+    d.print(((MODES[MODE] + ' ') if MODE != DISPLAY else '') + bs.decode(), \
+        fg = conf.fgcolor, bg = conf.bgcolor, posx = 0, posy = 7 * 8)
 
 def render(d):
     ltime = utime.localtime()
     years = ltime[0]
     months = ltime[1]
     days = ltime[2]
-    hours = ltime[3]
-    mins = ltime[4]
     secs = ltime[5]
 
-    d.clear()
+    d.clear(col = conf.bgcolor)
 
-    if MODE == CHANGE_YEAR:
-        renderNum(d, years // 100, 1)
-        renderNum(d, years % 100, 13)
-    elif MODE == CHANGE_MONTH:
-        renderNum(d, months, 13)
-    elif MODE == CHANGE_DAY:
-        renderNum(d, days, 13)
-    else:
-        renderNum(d, hours, 1)
-        renderNum(d, mins, 13)
+    theme.renderTime(d, ltime, MODE, conf.fgcolor)
 
-    if MODE not in (CHANGE_YEAR, CHANGE_MONTH, CHANGE_DAY) and secs % 2 == 0:
-        renderColon(d)
+    if MODE == DISPLAY and int(secs / 10) % 2 == 0:
```

Comment:
```suggestion
    if MODE == DISPLAY and secs // 10 % 2 == 0:
```

Python has floor divide for this exact reason. It's probably a good idea to use this.

Reply:
Ah, yes! Didn't really know about that feature before, that is better.